### DISCUSSION TITLE
resizing the native window in NiceGui

### DISCUSSION THREAD

**Discussion by impubes** [zauberzeug/nicegui#909] [Q&A] [answered]
### Question

Hi everyone,
I am currently experimenting with NiceGui and trying to resize the native window during runtime. There is "pywebview.api.changesize()", but I'm not sure how to use it (or maybe it is not possible to use it i dont know) in NiceGui. Could someone please provide me with some guidance or examples? 
Thank you in advance for your help!

**Comment by rodja** [comment]
Thanks for reaching out @impubes. Currently there is no way to resize the native window. pywebview is running in a different process than the FastAPI server (and the NiceGUI event loop). But I've created the pull-request #917 which uses a proxy object to do it anyway. It's quite complicated code, but should be very easy to use. Would you like to try it and tell me if it is working for you?

**Reply by impubes** [reply]
Thank you @rodja

**Comment by falkoschindler** [comment, marked as answer]
Since NiceGUI 1.2.16 you can manipulate the native window like this:

```py
app.native.main_window.resize(1000, 700)
```

See PR #917 for more information.

**Reply by impubes** [reply]
Thank you @falkoschindler nice to see it is implemented in new version 🥳
